src/main/java/com/umc/networkingService/domain/schedule/service/ScheduleServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+import com.umc.networkingService.domain.member.entity.Member;
+import com.umc.networkingService.domain.schedule.dto.request.ScheduleRequest.CreateSchedule;
+import com.umc.networkingService.domain.schedule.dto.request.ScheduleRequest.UpdateSchedule;
+import com.umc.networkingService.domain.schedule.dto.response.ScheduleResponse.ScheduleDetail;
+import com.umc.networkingService.domain.schedule.dto.response.ScheduleResponse.ScheduleId;
+import com.umc.networkingService.domain.schedule.dto.response.ScheduleResponse.ScheduleInfoSummariesInCalendar;
+import com.umc.networkingService.domain.schedule.dto.response.ScheduleResponse.ScheduleInfoSummary;
+import com.umc.networkingService.domain.schedule.dto.response.ScheduleResponse.ScheduleInfoSummaryLists;
+import com.umc.networkingService.domain.schedule.entity.Schedule;
+import com.umc.networkingService.domain.schedule.mapper.ScheduleMapper;
+import com.umc.networkingService.domain.schedule.repository.ScheduleRepository;
+import com.umc.networkingService.global.common.exception.ErrorCode;
+import com.umc.networkingService.global.common.exception.RestApiException;
+import java.util.List;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ScheduleServiceImpl implements ScheduleService {
+    private final ScheduleRepository scheduleRepository;
+    private final ScheduleMapper scheduleMapper;
+
+    @Override
+    public ScheduleInfoSummariesInCalendar getCalendarByMonth(Long month) {
+
+
+        return scheduleMapper.toScheduleInfoSummariesInCalendar(
+                scheduleRepository.findSchedulesByMonth(month).stream()
+                .map(schedule -> scheduleMapper.toScheduleInfoSummaryInCalendar(schedule))
+                .toList());
+    }
+
+    @Override
+    public ScheduleInfoSummaryLists getScheduleLists(Long month) {
+        List<Schedule> schedulesLists = scheduleRepository.findSchedulesByMonth(month);
+        List<ScheduleInfoSummary> campusSchedules = schedulesLists.stream()
+                .filter(schedule -> schedule.getHostType().equals(HostType.CAMPUS))
+                .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
+                .toList();
+
+        List<ScheduleInfoSummary> branchSchedules = schedulesLists.stream()
+                .filter(schedule -> schedule.getHostType().equals(HostType.BRANCH))
+                .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
+                .toList();
+
+        List<ScheduleInfoSummary> centerSchedules = schedulesLists.stream()
+                .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
+                .filter(schedule -> schedule.getHostType().equals(HostType.CENTER))
+                .toList();
+
+        return scheduleMapper.toScheduleInfoSummaries(campusSchedules, branchSchedules, centerSchedules);
+    }
```

Comment:
코드가 중복이 너무 많은데 다음과 같이 바꿔보는 것은 어떨까요?
```suggestion
@Override
public ScheduleInfoSummaryLists getScheduleLists(Long month) {
    List<Schedule> schedulesLists = scheduleRepository.findSchedulesByMonth(month);

    Map<HostType, List<ScheduleInfoSummary>> schedulesByType = schedulesLists.stream()
            .map(schedule -> scheduleMapper.toScheduleInfoSummary(schedule))
            .collect(Collectors.groupingBy(ScheduleInfoSummary::getHostType));

    return scheduleMapper.toScheduleInfoSummaries(
            schedulesByType.getOrDefault(HostType.CAMPUS, Collections.emptyList()),
            schedulesByType.getOrDefault(HostType.BRANCH, Collections.emptyList()),
            schedulesByType.getOrDefault(HostType.CENTER, Collections.emptyList())
    );
}
```